src/htm/encoders/RandomDistributedScalarEncoder.cpp

```diff
@@ -118,14 +120,30 @@ void RandomDistributedScalarEncoder::encode(Real64 input, SDR &output)
     // deviations in the sparsity or semantic similarity, depending on how
     // they're handled.
 
-    // Exercise for the reader: Calculate the probability of a hash collision
-    // and account for it in the sparsity.
-
     data[bucket] = 1u;
   }
   output.setDense( data );
 }
 
+bool RandomDistributedScalarEncoder::check_parameters() {
+  if( parameters.size >= 1000 && parameters.activeBits >= 10 ) {
+    return true;
+  }
+  const auto maximum_overlap = 0.33f;
+  const auto number_of_trials = 1000u;
```

Comment:
nit: this could have been default params to the constructor, or just set in init as `self.max_overlap`,..

Reply:
Hi Breznak,

Yes but I disagree for two reasons:

1) I tuned the heuristic to skip this method manually by trial and error. If the user changes the `maximum_overlap` then this heuristic might not be valid anymore (it would allow the parameters even though there is a chance that they are bad).

2) I don't think it would be useful for the users. My goal here is just to sanity check the users input for obviously bad parameters. I think that if the users want to control the `maximum_overlap` between two RDSE representations that they would be better off spending some time playing around with SDR's in order to gain an intuitive understanding how SDR size & sparsity affect the probability of collisions between random SDRs.